lti_consumer/lti_1p3/consumer.py

```diff
@@ -625,6 +721,173 @@ def check_and_decode_deep_linking_token(self, token):
         # Return contentitems
         return content_items
 
+    def _check_token_claim(self, token, claim_key, expected_value, invalid_claim_error_msg):
+        claim_value = token.get(claim_key)
+
+        if claim_value is None:
+            raise exceptions.MissingRequiredClaim(f"Token is missing required {claim_key} claim.")
+        if claim_value != expected_value:
+            raise exceptions.InvalidClaimValue(invalid_claim_error_msg)
+
+    def check_and_decode_proctoring_token(self, token):
+        """
+        Check and decode Proctoring LtiStartAssessment response and return relevant data to the Assessment Platform.
+        """
+        if not self.proctoring:
+            raise exceptions.LtiAdvantageServiceNotSetUp()
+
+        # Decode token and check expiration.
+        proctoring_response = self.tool_jwt.validate_and_decode(token)
+
+        # TODO: We MUST perform other forms of validation here.
+        # TODO: We MUST validate the verified_user claim if it is provided, although it is optional to provide it.
+        #       An Assessment Platform MAY reject the Start Assessment message if a required identity claim is missing
+        #       (indicating that it has not been verified by the Proctoring Tool).  Which identity claims a
+        #       Proctoring Tool will verify is subject to agreement outside of the scope of this specification but, at a
+        #       minimum, it is recommended that Proctoring Tools performing identity verification are able to verify the
+        #       given_name, family_name and name claims.
+        #       See 3.3 Transferring the Candidate Back to the Assessment Platform.
+
+        # Check Required LTI Claims
+        # -------------------------
+
+        # Check that the response message_type claim is "LtiStartAssessment".
+        # message_type = proctoring_response.get("https://purl.imsglobal.org/spec/lti/claim/message_type")
+        # if message_type is None:
+        #     raise exceptions.MissingRequiredClaim("Token is missing required message_type claim.")
+        # if message_type != "LtiStartAssessment":
+        #     raise exceptions.InvalidClaimValue("Token's message_type claim should be LtiStartAssessment.")
+
+        claim_key = "https://purl.imsglobal.org/spec/lti/claim/message_type"
+        self._check_token_claim(
+            proctoring_response,
+            claim_key,
+            "LtiStartAssessment",
+            f"Token's {claim_key} claim should be LtiStartAssessment."
+        )
+
+        # Check that the response version claim is "1.3.0".
```

Comment:
The next few checks are generic to all messages received from the tool (namely `LtiDeepLinkingResponse`) - so maybe this should be generic?